account_bank_statement_import_adyen/models/account_bank_statement_import.py

```diff
@@ -52,6 +52,11 @@ def import_adyen_transaction(self, statement, statement_id, row):
             amount=self.balance(row),
             note='%s %s %s %s' % (row[2], row[3], row[4], row[21]),
             name="%s" % (row[3] or row[4] or row[9]),
+            partner_id=self.env['account.move.line'].search([
+                '|',
+                ('move_id.name', '=', row[3] or row[4] or row[9]),
+                ('name', '=', row[3] or row[4] or row[9]),
```

Comment:
so here I'm cheaping out a little bit, Odoo standard would be only `move_id.name`, with this it works for us and for Odoo standard. Acceptable? Otherwise, we'd either need a private override for this or a migration that moves all psp numbers into the move name, but possibly we want to keep the move name in sync with the sequence?

Reply:
This is alright, but can you add an extra filter on receivable move lines (should help performance) and check the amount (which should always match)?

Reply:
I also added a few other restrictions